app/src/main/java/com/example/plantsapp/presentation/ui/MainActivity.kt

```diff
@@ -14,12 +15,29 @@ import dagger.hilt.android.AndroidEntryPoint
 @AndroidEntryPoint
 class MainActivity : AppCompatActivity() {
 
+    private val binding: ActivityMainBinding by lazy {
+        ActivityMainBinding.inflate(layoutInflater)
+    }
+    // TODO maybe find a more proper way to handle bottom navigation visibility
+    private val fragmentLifecycleCallback = object : FragmentManager.FragmentLifecycleCallbacks() {
+        override fun onFragmentStarted(fm: FragmentManager, f: Fragment) {
+            super.onFragmentStarted(fm, f)
+            if (f is TasksForDaysFragment) {
+                binding.bottomNavigation.isVisible = true
+            }
```

Comment:
1. Process both states
2. Watch in debugger fm.fragments  list to see how they are stored , and use last or first of this list to decide should this view be shown or not 
3. It is better to hide bottom bar in specific cases and show in all other cases

Reply:
So it should look something like this:
binding.bottomNavigation.isVisible = currentVisibleFragment !is AuthFragment